### DISCUSSION TITLE
how can i find tasmota device from mobile

### DISCUSSION THREAD

**Discussion by kareemkhalil** [arendst/Tasmota#15808] [Ideas]
Hello Dears, 
i am using Tasmota in my iot device i am now implementing an android app to control tasmota devices through it, what i want to ask i want to make a feature for user to search automatically for tasmota devices on the network and add it. 

how can i handle this from code?

**Comment by Jason2866** [comment]
Sorry this question is to complex to answer in discussions. 
Short answer. Many changes needed which will probably not added when a PR is submitted.
-> Not a design goal of Tasmota

**Reply by sfromis** [reply]
The Android device could subscribe to MQTT, which would give it access to retained messages with details about all Tasmota devices having subscribed.

**Comment by lalo-uy** [comment]
You can do a nmap of the subnet looking for devices with port 80 open.
Then send status cmnd and see the result.

**Reply by sfromis** [reply]
Such command line tools may not be permitted in a standard Android setup.

**Reply by MaceMoneta** [reply]
You can use the WiFiMan app in the Play Store (writen by Ubiquiti):

https://play.google.com/store/apps/details?id=com.ubnt.usurvey

On the Discovery tab, search for espressif, and it will find all esp8266  / esp32 devices.
